cdap-ui/app/cdap/components/TextboxOnValium/index.js

```diff
@@ -53,6 +53,18 @@ export default class TextboxOnValium extends Component {
       return;
     }
 
+    if (
+      textValue.length > 0 &&
+      !isNil(this.props.invalidCharacterRegex) &&
+      this.props.invalidCharacterRegex instanceof RegExp &&
+      this.props.invalidCharacterRegex.test(textValue)
```

Comment:
nit: Seems like 3 of the 4 conditions is common to the if structure above. Can we extract that out to to a boolean variable and use that?

Reply:
Also shouldn't we check for the `invalidReplacementCharacter` validity too?